src/py21cmfast/src/SpinTemperatureBox.c

```diff
@@ -1,2760 +1,1932 @@
-                        sigma_Tmin[R_ct] = sigma_z0(fmaxf(M_MIN, M_MIN_WDM));
-                    }
-                    else {
-                        sigma_Tmin[R_ct] = sigma_z0(fmaxf((float)TtoM(zpp, astro_params->X_RAY_Tvir_MIN, mu_for_Ts), M_MIN_WDM));
+    calculate_spectral_factors(zp);
+
+    //Fill the R_ct,box_ct fields
+    //Since we use the average Mturn for the global tables this must be done first
+    //NOTE: The filtered Mturn for the previous snapshot is used for Fcoll at ALL zpp
+    //  regardless of distance from current reshift, this also goes for the averages
+    //NOTE: Won't the average Mturn be the same for all R, since its just filtered?
+    //TODO: These R_ct x box_ct grids are the subsitute for XraySourcebox for the no-halo case
+    //  It should be more efficient to replace it with that structure in future, simply calculated
+    //  from the density grid at one redshift (or maybe implement the annular filtering there too)
+    //  This will involve the function which computes the SFRD from filtered density and Mcrit grids
+    double ave_log10_MturnLW[global_params.NUM_FILTER_STEPS_FOR_Ts];
+    double min_log10_MturnLW[global_params.NUM_FILTER_STEPS_FOR_Ts];
+    double max_log10_MturnLW[global_params.NUM_FILTER_STEPS_FOR_Ts];
+    double ave_dens[global_params.NUM_FILTER_STEPS_FOR_Ts];
+    fftwf_complex *log10_Mcrit_LW_unfiltered;
+    fftwf_complex *delta_unfiltered;
+    double log10_Mcrit_mol, curr_vcb;
+    double max_buf=-1e20, min_buf=1e20, curr_dens;
+    curr_vcb = flag_options_ts->FIX_VCB_AVG ? global_params.VAVG : 0;
+
+    //TODO: I want to move this part of the box assignment to an XraySourceBox for consistency between
+    //  halo/nohalo flags and options to use the proper perturbfield/SFRD and annular filters
+    if(!flag_options->USE_HALO_FIELD){
+        //copy over to FFTW, do the forward FFTs and apply constants
+        delta_unfiltered = (fftwf_complex *)fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
+        if(flag_options->USE_MINI_HALOS)
+            log10_Mcrit_LW_unfiltered = (fftwf_complex *)fftwf_malloc(sizeof(fftwf_complex)*HII_KSPACE_NUM_PIXELS);
+
+        prepare_filter_boxes(redshift,perturbed_field->density,ini_boxes->lowres_vcb,previous_spin_temp->J_21_LW_box,
+                            delta_unfiltered,log10_Mcrit_LW_unfiltered);
+        LOG_DEBUG("Forward FFT Done");
+        //fill the filtered boxes if we are storing them all
+        if(!user_params->MINIMIZE_MEMORY){
+            fill_Rbox_table(delNL0,delta_unfiltered,R_values,global_params.NUM_FILTER_STEPS_FOR_Ts,-1,inverse_growth_factor_z,min_densities,ave_dens,max_densities);
+            if(flag_options->USE_MINI_HALOS){
+                //NOTE: we are using previous_zp LW threshold for all zpp, inconsistent with the halo model
+                log10_Mcrit_mol = log10(lyman_werner_threshold(zp, 0., 0.,astro_params)); //minimum turnover NOTE: should be zpp?
```

Comment:
I believe so, it was written like this to reproduce previous results. There is not much of an effect since all this does is set a lower limit on the filtered Mcrit grids. I will replace the minimum and constant factor in `fill_rbox_table` to take separate values for each redshift/filter radius